Sources/Networking/Operations/PostOfferForSigningOperation.swift

```diff
@@ -79,26 +81,34 @@ class PostOfferForSigningOperation: NetworkOperation {
     }
 
     private static func handleOffer(
-        _ offer: [String: Any],
+        _ offer: PostOfferResponse.Offer,
         statusCode: HTTPStatusCode
     ) -> Result<PostOfferForSigningOperation.SigningData, BackendError> {
-        if let signatureError = offer["signature_error"] as? [String: Any] {
+        if let signatureError = offer.signatureError {
             return .failure(
-                .networkError(.errorResponse(ErrorResponse.from(signatureError), statusCode))
+                .networkError(.errorResponse(signatureError, statusCode))
             )
-        } else if let signatureData = offer["signature_data"] as? [String: Any],
-                  let signature = signatureData["signature"] as? String,
-                  let keyIdentifier = offer["key_id"] as? String,
-                  let nonce = (signatureData["nonce"] as? String).flatMap({ UUID(uuidString: $0) }),
-                  let timestamp = signatureData["timestamp"] as? Int {
-            return .success((signature, keyIdentifier, nonce, timestamp))
+        } else if let signingData = offer.asSigningData {
+            return .success(signingData)
         } else {
-            return .failure(.unexpectedBackendResponse(.postOfferIdSignature, extraContext: offer.stringRepresentation))
+            return .failure(
+                .unexpectedBackendResponse(.postOfferIdSignature, extraContext: "\(offer)")
+            )
         }
     }
 
 }
 
+private extension PostOfferResponse.Offer {
+
+    var asSigningData: PostOfferForSigningOperation.SigningData? {
+        guard let signature = self.signatureData else { return nil }
+
+        return (signature.signature, self.keyIdentifier, signature.nonce, signature.timestamp)
```

Comment:
`signature.signature` weird, but like, I get it.

Reply:
Good point, maybe I'll rename this `signature` to `data`.

Reply:
Oh that's better, ya